corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -160,6 +161,30 @@ def _process_main_forms(self):
         while not pool.join(timeout=10):
             self.log_info('Waiting on {} docs'.format(len(pool)))
 
+    def _try_to_process_form(self, wrapped_form, pool):
+        case_ids = get_case_ids_from_form(wrapped_form)
+        if self.queues.try_obj(case_ids, wrapped_form):
+            pool.spawn(self._migrate_form_and_associated_models_async, wrapped_form)
+        elif self.queues.full:
+            sleep(0.01)  # swap greenlets
+
+    def _try_to_process_queues(self, pool):
+        # regularly check if we can empty the queues
+        while True:
+            new_wrapped_form = self.queues.get_next()
+            if not new_wrapped_form:
+                break
+            pool.spawn(self._migrate_form_and_associated_models_async, new_wrapped_form)
+
+    def _rebuild_queues(self, pool):
+        prev_ids = self.queues.get_ids_from_run_timestamp()
+        form_db = dbaccessors.FormAccessors(self.domain)
+        for form_id in prev_ids:
```

Comment:
efcfd07

Reply:
It turns out that it's on the sql overridden thread at this point, so db accessors directed it to the sql db. I switched this to use FormAccessorCouch, which doesn't have an iter_forms. I don't think it should matter speed-wise though, since there should never be _too_ many things in the queue.